### PR TITLE
Hide "retry" button on results screen after watching a replay

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on https://github.com/ppy/osu/pull/8488

### CODE REVIEW

**Comment by TheWildTree** [comment]
Just a suggestion - I kind of feel that while quick-retry during a replay feels weird, the retry (or here it should say... replay?) button wouldn't be so bad to stay in this case - having a quick way to rewatch a replay after it's ended feels like a much better UX than having to exit, go to the leaderboards, sometimes scroll and find the same replay and only then be able to watch it again.

**Comment by peppy** [comment]
There's a button to watch the replay again. It's the "watch replay" button.

**Comment by TheWildTree** [comment]
Oops, I confused those two. My bad.

File: osu.Game/Screens/Ranking/ResultsScreen.cs [hunk 1]
```diff
@@ -87,15 +91,20 @@ private void load()
 
             if (player != null)
             {
-                AddInternal(new HotkeyRetryOverlay
+                if (allowRetry)
```
Comment (bdach) [inline osu.Game/Screens/Ranking/ResultsScreen.cs]: Maybe collapse these conditionals into one or factor out a method for adding the retry stuff? It's a bit nested right now.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**